docs/using/syntax-optional.md

```diff
@@ -1,32 +1,202 @@
+
+`www.example.com` -> www.example.com
+
+:::{important}
+This extension requires that [linkify-it-py](https://github.com/tsutsu3/linkify-it-py) is installed.
+Either directly; `pip install linkify-it-py` or *via* `pip install myst-parser[linkify]`.
+:::
+
+(syntax/substitutions)=
+
+## Substitutions
+
+Adding `"substitution"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will allow you to add substitutions, added in either the `conf.py` using `myst_substitutions`:
+
+```python
+myst_substitutions = {
+  "key1": "I'm a **substitution**"
+}
+```
+
+or at the top of the file, in the front-matter section (see [this section](syntax/frontmatter)):
+
+````yaml
+---
+substitutions:
+  key1: "I'm a **substitution**"
+  key2: |
+    ```{note}
+    {{ key1 }}
+    ```
+  key3: "![fun fish](img/fun-fish.png)"
+---
+````
+
+:::{important}
+Keys in the front-matter will override ones in the `conf.py`.
+:::
 
-A special syntax for admonitions can optionally be enabled by setting `myst_admonition_enable = True` in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html).
+You can use these substitutions inline or as blocks, and you can even nest substitutions in other substitutions:
 
-The key differences are that, instead of back-ticks, colons are used, and **the content starts as regular Markdown**.
-This has the benefit of allowing the content to be rendered correctly, when you are working in any standard Markdown editor.
-For example:
+:::{tabbed} Markdown Input
 
+```md
+Inline: {{ key1 }}
+
+{{ key2 }}
+
+| col1     | col2     |
+| -------- | -------- |
+| {{key3}} | {{key3}} |
+
+```
+
+:::
+
+:::{tabbed} Rendered Output
+Inline: {{ key1 }}
+
+{{ key2 }}
+
+| col1     | col2     |
+| -------- | -------- |
+| {{key3}} | {{key3}} |
+
+:::
+
+:::{important}
+
+Substitutions will only be assessed where you would normally use Markdown, e.g. not in code blocks:
+
+````
+```
+{{ key1 }}
+```
+````
+
+```
+{{ key1 }}
+```
+
+:::
+
+(syntax/colon_fence)=
+
+## Code fences using colons
+
+By adding `"colon_fence"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)),
+you can also use `:::` delimiters to denote code fences, instead of ```` ``` ````.
+
+Using colons instead of back-ticks has the benefit of allowing the content to be rendered correctly, when you are working in any standard Markdown editor.
+It is ideal for admonition type directives (as documented in [Directives](syntax/directives)) or tables with titles, for example:
+
+````{tabbed} Markdown Input
 ```md
 :::{note}
 This text is **standard** _Markdown_
 :::
+
+:::{table} This is a **standard** _Markdown_ title
+:align: center
+:widths: grid
+
+abc | mnp | xyz
+--- | --- | ---
+123 | 456 | 789
+:::
 ```
 
+````
+
+````{tabbed} Rendered Output
```

Comment:
just a note that I guess you wouldn't need to have four ` ```` ` in there if it's only nesting some fences `:::` right? E.g. you could do:

````
```
:::{note}
:::
```
````

Reply:
yep 👍